Examples/Sources/MacroExamples/Implementation/Member/MetaEnumMacro.swift

```diff
@@ -22,18 +22,22 @@ public struct MetaEnumMacro {
   let parentParamName: TokenSyntax
 
   init(node: AttributeSyntax, declaration: some DeclGroupSyntax, context: some MacroExpansionContext) throws {
-    guard let enumDecl = declaration.as(EnumDeclSyntax.self) else {
+    guard var enumDecl = declaration.as(EnumDeclSyntax.self) else {
       throw DiagnosticsError(diagnostics: [
         CaseMacroDiagnostic.notAnEnum(declaration).diagnose(at: Syntax(node))
       ])
     }
 
-    parentTypeName = enumDecl.name.with(\.trailingTrivia, [])
+    enumDecl.name.trailingTrivia = []
+
+    parentTypeName = enumDecl.name
```

Comment:
Since we don’t actually want to modify the `enumDecl` here, I think `with` reads nicer. Because that’s really what we are doing. Setting the `parentTypeName` to the `enumDecl`’s name but without trailing trivia.